### PR TITLE
docs(dbt): add reference to custom auto materialize policies / freshness policies

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation
https://github.com/dagster-io/dagster/pull/15480/files

### CODE REVIEW

**Review by rexledesma — APPROVED** [review]
Would recommend to have parallelism with the other sections that specify config on the dbt project yaml

File: docs/content/integrations/dbt/reference.mdx [hunk 1]
```diff
@@ -141,6 +141,8 @@ In Dagster, each asset definition has attributes. Dagster automatically generate
 - [Customizing group names](#customizing-group-names)
 - [Customizing descriptions](#customizing-descriptions)
 - [Customizing metadata](#customizing-metadata)
+- [Customizing auto materialize policies](#customizing-auto-materialize-policies)
```
Comment (erinkcochran87) [inline docs/content/integrations/dbt/reference.mdx:144]:
```suggestion
- [Customizing auto-materialize policies](#customizing-auto-materialize-policies)
```

The heading has this as hyphenated, but not here - adding it to be consistent

File: docs/content/integrations/dbt/reference.mdx [hunk 2]
```diff
@@ -304,6 +306,40 @@ def my_dbt_assets(context: OpExecutionContext, dbt: DbtCliResource):
     yield from dbt.cli(["build"], context=context).stream()
 ```
 
+### Customizing Auto-Materialize Policies
```
Comment (erinkcochran87) [inline docs/content/integrations/dbt/reference.mdx:309]:
```suggestion
### Customizing auto-materialize policies
```

Headings should use sentence case

File: docs/content/integrations/dbt/reference.mdx [hunk 3]
```diff
@@ -304,6 +306,40 @@ def my_dbt_assets(context: OpExecutionContext, dbt: DbtCliResource):
...
+        dagster:
+          auto_materialize_policy:
+            type: eager
+```
+
+### Customizing Freshness Policies
+
+For dbt models, seeds, and snapshots, the default <PyObject object="FreshnessPolicy"/> will be `None`.
+
+To add a FreshnessPolicy to a dbt node, you can define a `meta` key in your dbt project file, on your dbt node's property file, or on the node's in-file config block. This config accepts identical arguments to the <PyObject object="FreshnessPolicy"/> class. The following example applies a FreshnessPolicy for the following model:
```
Comment (erinkcochran87) [inline docs/content/integrations/dbt/reference.mdx:329]:
```suggestion
To add a `FreshnessPolicy` to a dbt node, you can define a `meta` key in your dbt project file, on your dbt node's property file, or on the node's in-file config block. This config accepts identical arguments to the <PyObject object="FreshnessPolicy"/> class. The following example applies a `FreshnessPolicy` for the following model:
```

Also, why does the second instance of `FreshnessPolicy` get a `PyObject`, but not the others? As you use it in the first sentence of this section, I don't think it's needed in this paragraph tbh

File: docs/content/integrations/dbt/reference.mdx [hunk 4]
```diff
@@ -304,6 +306,40 @@ def my_dbt_assets(context: OpExecutionContext, dbt: DbtCliResource):
...
+models:
+  - name: customers
+    config:
+      meta:
+        dagster:
+          auto_materialize_policy:
+            type: eager
+```
+
+### Customizing Freshness Policies
```
Comment (erinkcochran87) [inline docs/content/integrations/dbt/reference.mdx:325]:
```suggestion
### Customizing freshness policies
```

Same here, using sentence case in headings

File: docs/content/integrations/dbt/reference.mdx [hunk 5]
```diff
@@ -304,6 +306,40 @@ def my_dbt_assets(context: OpExecutionContext, dbt: DbtCliResource):
     yield from dbt.cli(["build"], context=context).stream()
 ```
 
+### Customizing Auto-Materialize Policies
+
+For dbt models, seeds, and snapshots, the default <PyObject object="AutoMaterializePolicy"/> will be `None`.
+
+To add an AutoMaterializePolicy to a dbt node, you can define a `meta` key in your dbt project file, on your dbt node's property file, or on the node's in-file config block. This policy may be one of two types, `"eager"` or `"lazy"`. The following example provides an eager AutoMaterializePolicy for the following model:
```
[the remark this answers is not in the document]
Comment (OwenKephart) [reply docs/content/integrations/dbt/reference.mdx:313]: ah interesting point -- it was intended to be the former (literal values), so I'll remove those quotes.

**Pull request merged**